### PR TITLE
MINOR: reduce() javadocs: clarify position of arguments

### PR DESCRIPTION
**Pull request by miguno**

### CODE REVIEW

**Comment by miguno** [comment]
Ping any of @guozhangwang @enothereska @dguy @mjsax

**Review by dguy — APPROVED** [review]
LGTM

File: streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedTable.java [hunk 1]
```diff
@@ -200,10 +201,11 @@
...
      * Each update to the original {@link KTable} results in a two step update of the result {@link KTable}.
      * The specified {@link Reducer adder} is applied for each update record and computes a new aggregate using the
-     * current aggregate and the record's value by adding the new record to the aggregate.
+     * current aggregate (first argument) and the record's value (second argument) by adding the new record to the
+     * aggregate.
      * The specified {@link Reducer substractor} is applied for each "replaced" record of the original {@link KTable}
-     * and computes a new aggregate using the current aggregate and the record's value by "removing" the "replaced"
-     * record from the aggregate.
+     * and computes a new aggregate using the current aggregate (first argument) and the record's value (second
+     * argument)by "removing" the "replaced" record from the aggregate.
```
Comment (enothereska) [inline streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedTable.java]: )by -> ) by (space needed)

**Comment by enothereska** [comment]
LGTM

File: streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedTable.java (hunk 1 above)
Comment (miguno) [reply streams/src/main/java/org/apache/kafka/streams/kstream/KGroupedTable.java]: Fixed <blush>.

**Comment by mjsax** [comment]
I am wondering, if changing the JavaDocs is sufficient -- why not rename the parameters of the `Reducer` directly?

**Comment by miguno** [comment]
@mjsax: I was thinking about this, too.

My current understanding is that the `Reducer` interface (to me) comes across as a simple binary function, and most probably -- though not explicitly defined in a contract as such -- an associative binary function at that.  That said, if `Reducer` is used nowhere else we may want to be more explicit, as you said.

**Comment by mjsax** [comment]
When reworking the JavaDocs, I had the same thought that `Reducer` interface is a simple binary function and agnostic if the parameters are "raw values" of "aggregates". That was the reason why I did not change anything. But in our code base, we only use it with "currentAgg" and "newValue" and as we describe this explicitly in the JavaDocs, we could also change the interface directly IMHO.

But it's hard to judge how to keep a nice "balance" and not mess with clean abstractions...

**Comment by miguno** [comment]
What shall we do with this PR?  Ready to merge (as @dguy and @enothereska said)?

I suppose any changes to the underlying code -- such as suggested by @mjsax -- should be done in  a different PR.

**Comment by mjsax** [comment]
I think, it describe the current code, so it should be fine to merge. And just to clarify: I am still not sure, if we should change the code...

**Comment by miguno** [comment]
@mjsax: Is this good to go? See your last comment above.  Your reply wasn't clear. :)

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/5784/
Test PASSed (JDK 8 and Scala 2.12).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/5798/
Test PASSed (JDK 7 and Scala 2.11).

**Review by mjsax — APPROVED** [review]
LGTM :)

**Comment by miguno** [comment]
Ping @dguy or @guozhangwang for merging.

**Comment by dguy** [comment]
Thanks @miguno merged to Trunk

**Pull request closed without merging**